packages/e2e-tests/package.json

```diff
@@ -37,6 +37,11 @@
 		"puppeteer-testing-library": "^0.5.0",
 		"uuid": "^8.3.0"
 	},
+	"optionalDependencies": {
```

Comment:
Should these be peer or option? I wasn't sure what to do, as I know `puppeteer` could be a very big thing to _have_ to bring in. (At the same time, for the e2e-tests package, it kind of makes sense to need it.)

Reply:
Actually, I just moved these to peers. Let me know what you think

Reply:
We have been doing all we can to avoid having `puppeteer` in the `node_modules` because it downloads Chromium as part of the post-install process. `puppeteer-core` is more difficult to configure and integrate, but it lets us defer that painful moment until someone runs e2e tests for the first time:

https://github.com/WordPress/gutenberg/blob/ebb4a8a031061410040625faa801981579880498/packages/scripts/scripts/test-e2e.js#L31-L33

In the past, we would alias `puppeteer` with `puppeteer-core` but it wasn't reliable with npm 6 so changed the strategy. Do you know which package declares `puppeteer` as a peer dependency?

Reply:
Great thought, I was worried about that. Do you think `optionalDependencies` would solve that problem??

Here is the explainer:

```js
$ yarn explain peer-requirements pea0fd
➤ YN0000: @wordpress/e2e-tests@portal:../gutenberg/packages/e2e-tests. [62522] doesn't provide puppeteer, breaking the following requirements:

➤ YN0000: puppeteer-testing-library@npm:0.5.0 [f94aa] → * ✘
```

It looks like `puppeteer-testing-library` has a peer dep requesting puppeteer:

https://github.com/testing-library/pptr-testing-library/blob/860106be873fa6830c49e529ec1a0b2a65f9bbbd/package.json#L75

Reply:
> Great thought, I was worried about that. Do you think `optionalDependencies` would solve that problem??

I have never tried using `optionalDependencies` because it seems like it still installs dependencies according to https://docs.npmjs.com/cli/v8/configuring-npm/package-json#optionaldependencies. The difference seems to be that it doesn't fail the whole installation process when it is unable to satisfy this dependency for any reason.

> It looks like `puppeteer-testing-library` has a peer dep requesting puppeteer:

This library is maintained by @kevin940726 so let's see what he thinks.

Reply:
I think we should list them as `peerDependencies`, but also mark them as optional using [`peerDependenciesMeta`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependenciesmeta).

We'll deprecate `puppeteer-testing-library` once we [migrate to Playwright](https://github.com/WordPress/gutenberg/pull/38570), I'd recommend ignoring the warning for now if it's possible, or I can quickly publish a patch to fix that.